core/src/components/datetime/datetime.tsx

```diff
@@ -959,6 +968,21 @@ export class Datetime implements ComponentInterface {
     root.addEventListener('ionBlur', (ev: Event) => ev.stopPropagation());
   }
 
+  /**
+   * When the presentation is changed, all calendar content is recreated,
+   * so we need to re-init behavior with the new elements.
+   */
+  componentDidRender() {
+    if (this.prevPresentation === null) {
+      this.prevPresentation = this.presentation;
+    }
+
+    if (this.presentation === this.prevPresentation) { return; }
+    this.prevPresentation = this.presentation;
+    this.destroyListeners();
```

Comment:
`destroyListeners` never destroys the overlay listener. Can we add this? otherwise we will have multiple `willPresent` callbacks firing.